helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java

```diff
@@ -66,31 +85,46 @@ public synchronized void refresh(HelixDataAccessor accessor) {
 
     super.refresh(accessor);
 
-    if (_sourceDataType.equals(PropertyType.TARGETEXTERNALVIEW) && _propertyDataChangedMap
-        .get(HelixConstants.ChangeType.TARGET_EXTERNAL_VIEW)) {
-      long start = System.currentTimeMillis();
-      _propertyDataChangedMap.put(HelixConstants.ChangeType.TARGET_EXTERNAL_VIEW, false);
-      _targetExternalViewCache.refresh(accessor);
-      LOG.info("Reload " + _targetExternalViewCache.getExternalViewMap().keySet().size()
-          + " TargetExternalViews. Takes " + (System.currentTimeMillis() - start) + " ms");
-    }
-
-    if (_sourceDataType.equals(PropertyType.CURRENTSTATES) && _propertyDataChangedMap
-        .get(HelixConstants.ChangeType.CURRENT_STATE)) {
-      long start = System.currentTimeMillis();
-      _propertyDataChangedMap.put(HelixConstants.ChangeType.CURRENT_STATE, false);
-      Map<String, LiveInstance> liveInstanceMap = getLiveInstances();
-      _currentStateCache.refresh(accessor, liveInstanceMap);
-      LOG.info("Reload CurrentStates. Takes " + (System.currentTimeMillis() - start) + " ms");
+    switch (_sourceDataType) {
+    case TARGETEXTERNALVIEW:
+      if (_propertyDataChangedMap.get(HelixConstants.ChangeType.TARGET_EXTERNAL_VIEW)) {
+        long start = System.currentTimeMillis();
+        _propertyDataChangedMap.put(HelixConstants.ChangeType.TARGET_EXTERNAL_VIEW, false);
+        _targetExternalViewCache.refresh(accessor);
+        LOG.info("Reload " + _targetExternalViewCache.getExternalViewMap().keySet().size()
+            + " TargetExternalViews. Takes " + (System.currentTimeMillis() - start) + " ms");
+      }
+      break;
+    case CURRENTSTATES:
+      if (_propertyDataChangedMap.get(HelixConstants.ChangeType.CURRENT_STATE)) {
+        long start = System.currentTimeMillis();
+        _propertyDataChangedMap.put(HelixConstants.ChangeType.CURRENT_STATE, false);
+        Map<String, LiveInstance> liveInstanceMap = getLiveInstances();
+        _currentStateCache.refresh(accessor, liveInstanceMap);
+        LOG.info("Reload CurrentStates. Takes " + (System.currentTimeMillis() - start) + " ms");
+      }
+      break;
+    case CUSTOMIZEDVIEW:
+      if (_propertyDataChangedMap.get(HelixConstants.ChangeType.CUSTOMIZED_VIEW)
+          && !_aggregationType.isEmpty()) {
+        long start = System.currentTimeMillis();
+        _propertyDataChangedMap.put(HelixConstants.ChangeType.CUSTOMIZED_VIEW, false);
+        _customizedViewCache.refresh(accessor);
+        LOG.info("Reload " + _customizedViewCache.getCustomizedViewMap().keySet().size()
+            + " CustomizedView. Takes " + (System.currentTimeMillis() - start) + " ms");
+      }
+      break;
+    default:
+      break;
```

Comment:
I think we shouldn't throw an exception here. If you look at the original code (without my changes), we do not through an exception. Also since we do not have ExternalView case here, we just want to ignore cases where the propertyType is externalView.